popde/density_estimate.py

```diff
@@ -1,3 +1,128 @@
+class SimpleGaussianKernelDensityEstimation:
+    """
+    Kernel Density Estimation (KDE) using Gaussian Kernels
+
+    Parameters:
+    -----------
+    data : array-like
+        The input data for which KDE will be estimated.
+
+    bandwidth : float, optional (default=1.0)
+        Bandwidth parameter controlling the width of the kernels.
+
+    kernel : str, optional (default='gaussian')
+        The kernel function used for smoothing. Supported kernels: 'gaussian',
+
+    Attributes:
+    -----------
+    data : array-like
+        The input data for which KDE is estimated.
+
+    bandwidth : float
+        Bandwidth parameter controlling the width of the kernels.
+
+    kernel : str
+        The kernel function used for smoothing.
+
+    Methods:
+    --------
+    fit(data, bandwidth=None, kernel=None):
+        Fit the KDE to new data.
+
+    evaluate(points):
+        Evaluate the KDE at given data points.
+
+    Examples:
+    ---------
+    # Create a KDE instance and fit it to data
+    data = np.random.randn(100)
+    kde = SimpleGaussianKernelDensityEstimation(data)
+    kde.fit()
+
+    # Evaluate the KDE at new data points
+    new_data = np.linspace(-3, 3, 100)
+    density_values = kde.evaluate(new_data)
+    """
+
+    def __init__(self, data, bandwidth=1.0, kernel='gaussian'):
+        """
+        Initialize the KernelDensityEstimation object.
+
+        Parameters:
+        -----------
+        data : array-like
+            The input data for which KDE will be estimated.
+
+        bandwidth : float, optional (default=1.0)
+            Bandwidth parameter controlling the width of the kernels.
+            Larger bandwidths lead to smoother density estimates.
+
+        kernel : str, optional (default='gaussian')
+            The kernel function used for smoothing.
+        """
+        import numpy as np
+
+        self.data = np.asarray(data)
+        self.bandwidth = bandwidth
+        self.kernel = kernel
+
+    def fit(self, data=None, bandwidth=None, kernel=None):
```

Comment:
How is this different from the `__init__` method?  They both set the values of the data, bandwidth and kernel attributes.

Reply:
fit is removed now.